core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -199,7 +241,7 @@ V get()
       // resourceHolderList can't have nulls, so we'll use a null to signal that we need to create a new resource.
       final V poolVal;
       synchronized (this) {
-        while (!closed && resourceHolderList.size() == 0 && deficit == 0) {
+        while (!closed && (numLentResources == maxSize)) {
```

Comment:
I am not sure but it seems that the new field `numLentResources` is not really needed. You could probably use the `deficit` variable to achieve similar behaviour?

Reply:
It is needed as an alternative for resourceHolderList.size() which works for both lazy and eager initialization